### PR TITLE
RX - Prevent SPEKTRUM receivebuffer overflow

### PR DESCRIPTION
**Pull request by ProDrone**
To solve the same problem for SPEKTRUM as mentioned for SBUS here:
https://github.com/cleanflight/cleanflight/issues/1512#issuecomment-159267342

NOTE: The problem is not the main subject of issue #1512, but another observation.

__UNTESTED. I don't own Spektrum hardware, someone please test and report back...__

### CODE REVIEW

File: src/main/rx/spektrum.c [hunk 1]
```diff
@@ -43,6 +43,7 @@
 #define SPEKTRUM_1024_CHANNEL_COUNT 7
 
 #define SPEK_FRAME_SIZE 16
+#define SPEKTRUM_TIME_NEEDED_PER_FRAME 5000
```
Comment (ledvinap) [inline src/main/rx/spektrum.c]: This is actually interframe/iterbyte delay. Line must be at idle between packets.

File: src/main/rx/spektrum.c [hunk 2]
```diff
@@ -96,21 +97,25 @@ bool spektrumInit(rxConfig_t *rxConfig, rxRuntimeConfig_t *rxRuntimeConfig, rcRe
 static void spektrumDataReceive(uint16_t c)
 {
     uint32_t spekTime;
-    static uint32_t spekTimeLast, spekTimeInterval;
-    static uint8_t spekFramePosition;
+    static uint32_t spekTimeInterval;
```
Comment (ledvinap) [inline src/main/rx/spektrum.c]: no `static`here

File: src/main/rx/spektrum.c (hunk 1 above)
Comment (ProDrone) [inline src/main/rx/spektrum.c]: Inter-frame is what i thought. Inter-byte confuses me. Individual bytes don't have this amount of spacing.
It is for packet synchronization, to know when a new packet starts.
It was a hard coded time, so i made a define for it.
Maybe better to call it SPEKTRUM_NEEDED_FRAME_INTERVAL

File: src/main/rx/spektrum.c (hunk 2 above)
Comment (ProDrone) [inline src/main/rx/spektrum.c]: Thanks, you are right, does not need to be static.

File: src/main/rx/spektrum.c (hunk 1 above)
Comment (ledvinap) [inline src/main/rx/spektrum.c]: SBUS calculates timeout from start of frame, SPECTRUM from last byte received ...

**Comment by hydra** [comment]
merged

**Pull request closed without merging**